addons/mail/static/src/core/public_web/discuss_app_model.js

```diff
@@ -43,6 +48,15 @@ export class DiscussApp extends Record {
         return super.insert(...arguments);
     }
 
+    get hasSubCategories() {
+        // check if user is part of any channel in any subcategory
+        return this.store.discuss.allCategories
+            .filter((c) => c.id !== "all" && !c.isRootCategory)
+            .map((c) => [...c.threads])
+            .flat(1)
+            .some((t) => t.displayToSelf || t.isLocallyPinned);
+    }
```

Comment:
Looks like an extremely badly scaling getter.
To have caching, this should probably be a compute, with possibly many sub-computes.
And if possible, being computed on the "one" side of the relations.